kernel/signal.c

```diff
@@ -298,3 +308,11 @@ long myst_signal_clone(myst_thread_t* parent, myst_thread_t* child)
 done:
     return ret;
 }
+
+long myst_handle_host_signal(unsigned signum, mcontext_t* mcontext)
+{
+    siginfo_t siginfo = {0};
+    siginfo.si_code = SI_KERNEL;
```

Comment:
At least for certain SIGSEGV handler implementation, `siginfo->si_addr` is needed, for example, [.NET runtime's SIGSEGV handler](https://github.com/dotnet/runtime/blob/0761780988b93a4d1cc98dc5d4e2d84db29e33fd/src/coreclr/pal/src/exception/signal.cpp#L556). The .NET runtime's signal handlers or other signal handler implementations might require other information in `siginfo`. It's better to support signal/exception specific `siginfo` setting in the HW exception handling code, instead of only allow common `siginfo` setting here.